### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by t-tsuya**
This pull request adds unregister functionality for activity participants, improves the frontend UI for managing participants, seeds additional activities, and introduces automated backend tests. Key changes include new API endpoints, enhanced participant management in the UI, and robust test coverage to ensure reliability.

**Backend: API and Data Improvements**
- Added a new `DELETE /activities/{activity_name}/signup` endpoint to allow students to unregister from activities, including validation for already unregistered students.
- Refactored the activity data model to use an `INITIAL_ACTIVITIES` seed and provided a `reset_activities` function for test isolation and easier state management. [[1]](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00R8-R23) [[2]](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00R41-R89)

**Frontend: Participant Management**
- Updated the UI to display a list of participants for each activity, including an "unregister" (delete) button next to each participant, and wired up the frontend to call the new unregister API endpoint. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R49-R87) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R98-R103)
- Improved user feedback by centralizing message display logic and ensuring activity lists refresh after signup or unregister actions.

**Styling Enhancements**
- Enhanced the visual presentation of participants and activity cards, including new styles for participant lists and unregister buttons. [[1]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R66) [[2]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R78-R133)

**Testing and Documentation**
- Added comprehensive backend tests for activity signup and unregister logic, including edge cases for duplicate signups and missing activities, using pytest and FastAPI’s TestClient. [[1]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R1-R14) [[2]](diffhunk://#diff-eafc5698f9b6a7272e496525da3098f0fc56b9c19e4ccb6eaf54108bbac9f0ceR1-R102)
- Updated documentation and setup instructions to use a requirements file and include test running steps. [[1]](diffhunk://#diff-e1d5879ce3e222eb0cdd43051fb8c34c00796467acb091d0b3fa939ee3355cb5L15-R15) [[2]](diffhunk://#diff-e1d5879ce3e222eb0cdd43051fb8c34c00796467acb091d0b3fa939ee3355cb5L24-R30)

### CODE REVIEW

File: src/static/app.js [hunk 1]
```diff
@@ -12,19 +46,45 @@ document.addEventListener("DOMContentLoaded", () => {
...
         const spotsLeft = details.max_participants - details.participants.length;
+        const hasParticipants = details.participants.length > 0;
+        const participantsList = hasParticipants
+          ? details.participants
+              .map(
+                (participant) => `
+                  <li class="participant-item">
+                    <span>${participant}</span>
+                    <button
+                      type="button"
```
[the remark this answers is not in the document]
Comment (t-tsuya) [reply src/static/app.js]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/t-tsuya/skills-getting-started-with-github-copilot/pull/2#discussion_r2909244270)

**Comment by t-tsuya** [comment]
@copilot+claude-opus-4.6 open a new pull request to apply changes based on the comments in [this thread](https://github.com/t-tsuya/skills-getting-started-with-github-copilot/pull/2#pullrequestreview-3919653283)
